lib/github/api.nit

```diff
@@ -1602,13 +956,301 @@ end
 #
 # Mostly a wrapper around a json object.
 class GithubFile
-
-	# Json content.
-	var json: JsonObject
+	serialize
 
 	# File name.
-	fun filename: String do return json["filename"].as(String)
+	var filename: String
+end
 
-	# Set filename.
-	fun filename=(filename: String) do json["filename"] = filename
+# Make ISO Datew serilizable
+redef class ISODate
+	super Jsonable
+	serialize
+
+	redef fun to_json do return serialize_to_json
 end
+
+# JsonDeserializer specific for Github objects.
+class GithubDeserializer
+	super JsonDeserializer
+
+	redef fun class_name_heuristic(json_object) do
+		if json_object.has_key("login") or json_object.has_key("email") then
+			return "User"
+		else if json_object.has_key("full_name") then
+			return "Repo"
+		else if json_object.has_key("name") and json_object.has_key("commit") then
+			return "Branch"
+		else if json_object.has_key("sha") and json_object.has_key("ref") then
+			return "PullRef"
+		else if json_object.has_key("sha") or (json_object.has_key("id") and json_object.has_key("tree_id")) then
+			return "Commit"
+		else if json_object.has_key("number") and json_object.has_key("patch_url") then
+			return "PullRequest"
+		else if json_object.has_key("open_issues") and json_object.has_key("closed_issues") then
+			return "Milestone"
+		else if json_object.has_key("number") and json_object.has_key("title") then
+			return "Issue"
+		else if json_object.has_key("color") then
+			return "Label"
+		else if json_object.has_key("event") then
+			return "IssueEvent"
+		else if json_object.has_key("original_commit_id") then
+			return "ReviewComment"
+		else if json_object.has_key("commit_id") then
+			return "CommitComment"
+		else if json_object.has_key("issue_url") then
+			return "IssueComment"
+		end
+		return null
+	end
```

Comment:
It is: http://nitlanguage.org/doc/stdlib/class_json__serialization__JsonDeserializer.html#json__serialization__JsonDeserializer__class_name_heuristic.definition